### PR TITLE
Restructuring The Docs

### PR DESCRIPTION
**Pull request by Bentlybro**
Updating and Restructuring the docs.
people working on this: @Bentlybro @Yash-Ambekar @sohamtembhurne @BarneyChambers @ConTronTech

 - [x] Move all specific documentation about Auto-GPT (the agent) into a subfolder
 	- [x] Update setup & usage instructions to match the new repo structure
 - [x] Create subfolders with main pages for benchmark, forge, and front end
 - [x] Create new main page
 - [x] Update Installation page to make it cleaner

### CODE REVIEW

**Comment by Pwuts** [comment]
Notes / feedbacks:
* I'll take a shot at rewriting the intro for the AutoGPT Agent

* Pascal-Kebab-Case in navigation drawer -> Just Normal Capitalized Text?
  image
  
* https://deploy-preview-5441--auto-gpt-docs.netlify.app/benchmarks/benchmark/ <-- The results on this page are out of date afaik

* This (and any other similar "outcries") should be formatted as [admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/)
  image

File: docs/content/AutoGPT/Setups/Docker-setup.md [hunk 1]
```diff
@@ -1,37 +1,12 @@
...
-    You can set up a paid account at [Manage account > Billing > Overview](https://platform.openai.com/account/billing/overview).
-
-[openai/api limits]: https://platform.openai.com/docs/guides/rate-limits/overview#:~:text=Free%20trial%20users,RPM%0A40%2C000%20TPM
+### Set up with Docker
 
-!!! important
-    It's highly recommended that you keep track of your API costs on [the Usage page](https://platform.openai.com/account/usage).
-    You can also set limits on how much you spend on [the Usage limits page](https://platform.openai.com/account/billing/limits).
+!!! important "Docker Setup Issue"
+    The Docker setup is currently broken because of an issue with Poetry.
```
Comment (andrewhooker2) [inline docs/content/AutoGPT/Setups/Docker-setup.md]:
```suggestion
    We are addressing a known issue with the Docker setup related to Poetry.
```

File: docs/content/AutoGPT/Setups/Docker-setup.md [hunk 2]
```diff
@@ -1,37 +1,12 @@
...
-![For OpenAI API key to work, set up paid account at OpenAI API > Billing](./imgs/openai-api-key-billing-paid-account.png)
+    [**We have an open PR if you'd like to take a look**](https://github.com/python-poetry/poetry/issues/8548)
 
+    Please keep this in mind. We apologize for any inconvenience, and thank you for your patience.
 
-## Setting up AutoGPT
-
-### Set up with Docker
 
 1. Make sure you have Docker installed, see [requirements](#requirements)
```
Comment (andrewhooker2) [inline docs/content/AutoGPT/Setups/Docker-setup.md:11]: The link provided is a reference to the current page and will not navigate to a new page or section.

File: docs/content/AutoGPT/Setups/Git-setup.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    ```
+
+### Configuration
+
+1. Find the file named `.env.template` in the main `Auto-GPT` folder. This file may
+    be hidden by default in some operating systems due to the dot prefix. To reveal
+    hidden files, follow the instructions for your specific operating system:
+    [Windows][show hidden files/Windows], [macOS][show hidden files/macOS].
+2. Create a copy of `.env.template` and call it `.env`;
+    if you're already in a command prompt/terminal window: `cp .env.template .env`.
```
Comment (andrewhooker2) [inline docs/content/AutoGPT/Setups/Git-setup.md]:
```suggestion
    if you're already in a command prompt/terminal window: 
    ```shell
    cp .env.template .env
    ```
```

File: docs/content/AutoGPT/Setups/Git-setup.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+3. Open the `.env` file in a text editor.
+4. Find the line that says `OPENAI_API_KEY=`.
+5. After the `=`, enter your unique OpenAI API Key *without any quotes or spaces*.
```
Comment (andrewhooker2) [inline docs/content/AutoGPT/Setups/Git-setup.md]:
```suggestion
5. Insert your OpenAI API Key directly after = without quotes or spaces..
    ```yaml
    OPENAI_API_KEY=sk-qwertykeys123456
    ```
```

File: docs/content/AutoGPT/Setups/nogit-setup.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
+### Set up without Git/Docker
```
Comment (andrewhooker2) [inline docs/content/AutoGPT/Setups/nogit-setup.md]:
```suggestion
### Setting Up with Git/Docker
```

File: docs/content/AutoGPT/Setups/Git-setup.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    ```shell
+    cd AutoGPT/autogpts/autogpt
+    ```
+
+### Configuration
+
+1. Find the file named `.env.template` in the main `Auto-GPT` folder. This file may
+    be hidden by default in some operating systems due to the dot prefix. To reveal
+    hidden files, follow the instructions for your specific operating system:
+    [Windows][show hidden files/Windows], [macOS][show hidden files/macOS].
```
Comment (andrewhooker2) [inline docs/content/AutoGPT/Setups/Git-setup.md]:
```suggestion
    [Windows][show hidden files/Windows] and [macOS][show hidden files/macOS].
```

File: docs/content/index.md [hunk 7]
```diff
@@ -1,8 +1,59 @@
 # AutoGPT docs
 
-Welcome to AutoGPT. Please follow the [Installation](/setup/) guide to get started.
+Welcome to the AutoGPT Docs.
 
-!!! note
-    It is recommended to use a virtual machine/container (docker) for tasks that require high security measures to prevent any potential harm to the main computer's system and data. If you are considering to use AutoGPT outside a virtualized/containerized environment, you are *strongly* advised to use a separate user account just for running AutoGPT. This is even more important if you are going to allow AutoGPT to write/execute scripts and run shell commands!
+The AutoGPT project has been divided into four distinct sections namely [Agents](#welcome-to-the-heart-of-autogpt), [Benchmarks](#autogpt-benchmarks), [Forge](#autogpt-forge) and [Frontend](#frontend)
```
Comment (andrewhooker2) [inline docs/content/index.md]:
```suggestion
The AutoGPT project has been divided into four distinct sections: [Agents](#welcome-to-the-heart-of-autogpt), [Benchmarks](#autogpt-benchmarks), [Forge](#autogpt-forge), and [Frontend](#frontend)
```

File: docs/content/index.md [hunk 8]
```diff
@@ -1,8 +1,59 @@
...
 
-It is for these reasons that executing python scripts is explicitly disabled when running outside a container environment.
+---
+
+## [Welcome to the heart of AutoGPT](AutoGPT/setup.md)
+
+Where the magic happens – the AutoGPT Agent. This is your go-to toolkit for supercharging agents and AI innovation.
```
Comment (andrewhooker2) [inline docs/content/index.md]:
```suggestion
Welcome to the heart of AutoGPT, where the magic happens – the AutoGPT Agent. This is your go-to toolkit for supercharging agents and AI innovation.
```

File: docs/content/index.md [hunk 9]
```diff
@@ -1,8 +1,59 @@
...
+
+Forge your future with AutoGPT! The Forge is your innovation lab.
+
+- 🏗️ **Building with Ease** - We've set the groundwork so you can focus on your agent's personality and capabilities. Comprehensive tutorials are available [Here](https://aiedge.medium.com/autogpt-forge-e3de53cc58ec).
+
+---
+
+## [Frontend](front-end/the-ui.md)
+
+Harness your AI journey with the AutoGPT Front-End.
```
Comment (andrewhooker2) [inline docs/content/index.md]: Might be good to keep FrontEnd consistent from what was said earlier. 

```suggestion
Harness your AI journey with the AutoGPT FrontEnd.
```

File: docs/content/index.md [hunk 10]
```diff
@@ -1,8 +1,59 @@
...
+
+- 🎮 **User-Friendly Interface** - Manage your agents effortlessly.
+
+- 🔄 **Seamless Integration** - Smooth connectivity between your agent and our benchmarking system.
+
+---
+
+[Join the AutoGPT discord server for any queries](discord.gg/autogpt)
```
Comment (andrewhooker2) [inline docs/content/index.md]:
```suggestion
[Join the AutoGPT Discord server for any queries](discord.gg/autogpt)
```

**Review by andrewhooker2 — COMMENTED** [review]
### Hello! 👋

I have made some slight grammar and spelling checks in the comments below. 
I have also made some suggestions around verbiage for clarity. 

Overall your documentation is looking very nice! Great job on this!

**Comment by Bentlybro** [comment]
Hey @andrewhooker2 thank you for the help with this it means a lot 🙏

**Pull request merged**